### ISSUE TITLE
Flutter 3.7 Magnifier on TextField gets stuck when used on a scrollable view on iOS

### ISSUE THREAD

**Issue by js2702** [flutter/flutter#119711]
It seems that on iOS the magnifier of a TextField can get stuck in place after releasing the finger and it never disappears. This happens when you put it into a scrollable view. This only seems to happen when the TextField is empty.

**Steps to reproduce:**

Using an iOS device.
1. flutter run
2. Long press on the TextField with the text empty to trigger the magnifier
3. Scroll up with that same finger over the magnifier

The magnifier seems to get stuck in the same place.

**Expected results:**
No magnifier

**Actual results:** 

Stuck magnifier

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: ListView(
        children: [
          const Text(
            "Long press to trigger the magnifier without writing anything and drag upwards (like if you where to scroll at the same time) while doing it",
          ),
          const TextField(
            decoration: InputDecoration(hintText: "write something"),
          ),
          ...List.generate(
            20,
            (index) => ListTile(
              title: Text("Tile $index"),
            ),
          )
        ],
      ),
    );
  }
}
```

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 3.7.0, on macOS 13.1 22C65 darwin-x64, locale es-ES)
    • Flutter version 3.7.0 on channel stable at /Users/jorge/fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (hace 9 días), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/jorge/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (5 available)
...
[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

https://user-images.githubusercontent.com/4404515/216087106-d7fe5469-7bc3-4905-b9cb-2c1227d5db63.MP4

**Comment by anchaoLu** [comment]
same issue on iOS15.6

**Comment by anchaoLu** [comment]
only long press, don't move finger

**Comment by anchaoLu** [comment]
When use TextField as Scaffold appBar title
```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const TextField(),
        ),
        body: const Center(
          child: Text('textfield long press'),
        ),
      ),
    );
  }
}
```
after long press textfield, the menu bar offset too large
![IMG_0187](https://user-images.githubusercontent.com/106209730/216249143-7f60e75e-cae9-4bd6-adce-4cf14099a40b.PNG)

**Comment by muggleXin** [comment]
Had the same problem with version 3.7

**Comment by anchaoLu** [comment]
> It seems that on iOS the magnifier of a TextField can get stuck in place after releasing the finger and it never disappears. This happens when you put it into a scrollable view. This only seems to happen when the TextField is empty.
> 
> **Steps to reproduce:**
> 
> Using an iOS device.
> 
> 1. flutter run
> 2. Long press on the TextField with the text empty to trigger the magnifier
> 3. Scroll up with that same finger over the magnifier
> 
> The magnifier seems to get stuck in the same place.
> 
> **Expected results:** No magnifier
> 
> **Actual results:**
> 
> Stuck magnifier
> 
> ```dart
> import 'package:flutter/material.dart';
> 
> void main() {
>   runApp(const MyApp());
> }
> 
> class MyApp extends StatelessWidget {
>   const MyApp({super.key});
> 
>   // This widget is the root of your application.
>   @override
>   Widget build(BuildContext context) {
>     return MaterialApp(
>       title: 'Flutter Demo',
>       theme: ThemeData(
>         primarySwatch: Colors.blue,
>       ),
>       home: const MyHomePage(title: 'Flutter Demo Home Page'),
>     );
>   }
> }
> 
> class MyHomePage extends StatelessWidget {
>   const MyHomePage({super.key, required this.title});
> 
>   final String title;
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>       appBar: AppBar(),
>       body: ListView(
>         children: [
>           const Text(
>             "Long press to trigger the magnifier without writing anything and drag upwards (like if you where to scroll at the same time) while doing it",
>           ),
>           const TextField(
>             decoration: InputDecoration(hintText: "write something"),
>           ),
>           ...List.generate(
>             20,
>             (index) => ListTile(
>               title: Text("Tile $index"),
>             ),
>           )
>         ],
>       ),
>     );
>   }
> }
> ```
> 
> Logs
>  RPReplay_Final1675265403.MP4

If you only expected no magnifier, this code can help
```dart
TextField(
    magnifierConfiguration: TextMagnifierConfiguration.disabled,
),
```

**Comment by darshankawar** [comment]
Thanks for the report @js2702 
I tried the code sample on latest stable and do see the same behavior, ie, the magnifier gets stuck upon scrolling, but I had to input some text in order to bring up the magnifier.

I also verified the same on latest master and found out that the reported behavior does not occur in it. So I suggest you to switch to latest master and run your scenario again to see if it works at your end or not.

1. Stable:


https://user-images.githubusercontent.com/67046386/216262672-781ad463-3271-416b-b435-ddd4518478f8.MP4


2. Master:


https://user-images.githubusercontent.com/67046386/216262750-e70b9463-8979-49a5-a537-1c28a7ee1d7d.MP4

**Comment by js2702** [comment]
Just checked, it seems to work fine on latest Master.

**Comment by darshankawar** [comment]
Thanks for the update. I am going ahead and closing this as fixed since it works on latest master. Master fixes are rolled into stable once in a quarter, so hopefully this will be available in next stable release.

**Comment by smiera** [comment]
Confirm, bug is still in v3.7.6

**Comment by techouse** [comment]
The bug is still present in v3.7.7

**Comment by techouse** [comment]
The bug is no longer present in v3.7.8 🚀

**Comment by techouse** [comment]
The bug is still present in v3.7.8

**Comment by alienspaces** [comment]
@darshankawar Any chance this will make it into the v3.7.9 release? Have an app release pending a fix for this.

**Comment by darshankawar** [comment]
Everyone in this thread facing this issue. The fix will need to be cherry-picked in stable version. If anyone is interested, they can file a CP request using [this](https://github.com/flutter/flutter/issues/new?assignees=&labels=cp%3A+review&template=7_cherry_pick.yml&title=%5BCP%5D+%3Ctitle%3E) template, fill out requested details and then the team can decide if they want to cherry pick this fix in stable or not. Else, you'll need to wait till the fix lands in next stable release.

**Comment by akindone** [comment]
bofore the fix lands in the next stable release, we can use this code to disable magnifier globally.
```TextMagnifier.adaptiveMagnifierConfiguration = TextMagnifierConfiguration(
      shouldDisplayHandlesInMagnifier: false,
      magnifierBuilder: (
          BuildContext context,
          MagnifierController controller,
          ValueNotifier<MagnifierInfo> magnifierInfo,
          ) {
        switch (defaultTargetPlatform) {
          case TargetPlatform.iOS:
            return null;
          case TargetPlatform.android:
            return TextMagnifier(
              magnifierInfo: magnifierInfo,
            );
          case TargetPlatform.fuchsia:
          case TargetPlatform.linux:
          case TargetPlatform.macOS:
          case TargetPlatform.windows:
            return null;
        }
      }
  );```

**Comment by Painwraith** [comment]
Confirmed its not fixed as of 4/11/23 in Flutter 3.7.10 bug still exists.
